subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/impl/DefaultFileSystemSnapshotterTest.groovy

```diff
@@ -243,6 +248,39 @@ class DefaultFileSystemSnapshotterTest extends Specification {
         })
     }
 
+    def "snapshots a singletonFileTree as RegularFileSnapshot"() {
+        given:
+        def file = tmpDir.createFile('testFile')
+
+        when:
+        def tree = new FileTreeAdapter(new DefaultSingletonFileTree(file), TestFiles.patternSetFactory)
```

Comment:
Let's use also different ways from `FileOperations` to create those file trees to make sure we don't miss a corner case.

We should test:
- fileTree(file)
- configurableFiles(file).getAsFileTrees()
- the above, matching an include which matches/doesn't match the file.

Anything else I am missing?

Reply:
I don't understand how this tests SingletonFileTree as all of those above create DirectoryFileTrees as far as I understand. Don't they?

Reply:
`DirectoryFileTree`s are only created if the target is a directory. If the targets are regular files, then `DefaultSingletonFileTree` will be used.

Reply:
Added coverage for that